src/script/lua_api/l_localplayer.cpp

```diff
@@ -0,0 +1,410 @@
+/*
+Minetest
+Copyright (C) 2017 Dumbeldor, Vincent Glize <vincent.glize@live.fr>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "script/common/c_converter.h"
+#include "l_localplayer.h"
+#include "l_internal.h"
+
+LuaLocalPlayer::LuaLocalPlayer(LocalPlayer *m) {
+	m_localplayer = m;
+}
+
+void LuaLocalPlayer::create(lua_State *L, LocalPlayer *m) {
+	LuaLocalPlayer *o = new LuaLocalPlayer(m);
+	*(void **) (lua_newuserdata(L, sizeof(void *))) = o;
+	luaL_getmetatable(L, className);
+	lua_setmetatable(L, -2);
+
+	// Keep localplayer object stack id
+	int localplayer_object = lua_gettop(L);
+
+	lua_getglobal(L, "core");
+	lua_getfield(L, -1, "localplayer");
+	luaL_checktype(L, -1, LUA_TTABLE);
+	int uitable = lua_gettop(L);
+
+	lua_pushvalue(L, localplayer_object);
+	lua_setfield(L, uitable, "localplayer");
```

Comment:
Not sure if I'm reading this correctly but doesn't this make the object accessible at `core.localplayer.localplayer`? That doesn't seem like a good location for it.